python/ray/tests/test_advanced_9.py

```diff
@@ -264,6 +264,11 @@ def test_gcs_connection_no_leak(ray_start_cluster):
 
     ray.init(cluster.address)
 
+    # When we create a new node, the new raylet invokes RegisterGcs -> AsyncSubscribeAll
+    # -> AsyncGetAll -> GetAllJobInfo which makes the GCS to make a connection to each
+    # drivers. To make the connections consistent, pre-create the connection here.
+    ray._private.worker.global_worker.gcs_client.get_all_job_info()
+
```

Comment:
I still don't understand why we need to do this here?

Reply:
So I found after this PR, we got 1 more connection created in the test add node, and is not closed in test remove node. Then I found creating a node triggers gcs to connect with driver core workers, and that connection is persisted in core_worker_connection_pool in an LRU policy. Previously we did not have that because gcs had no way to connect to the driver (port is 0) which is exactly what we are fixing in this PR. So I pre-trigger the gcs->driver core worker connection to make it count in the initial list, so after the add node and remove node, the conn count is stable.